### ISSUE TITLE
Proposal: move components of `utils` into top-level and/or more descriptively-named packages

### ISSUE THREAD

**Issue by huonw** [stellargraph/stellargraph#909]
### Description

We currently have a `stellargraph.utils` package, that defines https://github.com/stellargraph/stellargraph/tree/8e3c9b64983d57afcdd7780f31de75436c7df2de/stellargraph/utils :

1. interpretability/saliency maps
2. calibration
3. ensembling
4. loss functions

I think we could consider moving this current set of things into other locations, e.g. top level packages.

One arrangement might be:

- `stellargraph.interpretability` or `stellargraph.importance` that contains saliency maps (and brute force interpretability: #850)
- `stellargraph.calibration`
- `stellargraph.ensembles`
- `stellargraph.losses` (mimicking `tf.keras.losses`)

Another organisation might be:

- `stellargraph.interpretability` or `stellargraph.importance` that contains saliency maps (and brute force interpretability: #850)
- `stellargraph.calibration`
- `stellargraph.model` that contains ensembles, as well as models like `GCN` (as in, separate the individual layers like `GraphConvolution` from the actual model like `GCN`)
- `stellargraph.losses`

The move can and should be done with appropriate deprecation warnings, so that any code using the old one continues working.

### Done Checklist

- [ ] Produced code for required functionality
- [ ] Tests written and coverage checked
- [ ] Code review performed
- [ ] Documentation on Google Docs  (if applicable)
- [ ] Documentation in repo
- [ ] Version number reflects new status
- [ ] CHANGELOG.md updated
- [ ] Team demo

**Comment by kjun9** [comment]
I think the idea behind the current `utils` package might've been as a collection of extra tools that can be used with any of the algorithms in the library? But yes, moving the inner modules to top-level sounds reasonable to me.

**Comment by huonw** [comment]
> I think the idea behind the current utils package might've been as a collection of extra tools that can be used with any of the algorithms in the library? 

Ah, so it was meant to serve as a "one stop shop" for that sort of thing? I guess importance, calibration and ensembles could all be moved into a single module (the only ideas I have for a name is something like `stellargraph.enrich` or `stellargraph.augment`, since these "enrich" or "augment" how the model works... but that's not very good).

**Comment by kjun9** [comment]
I was just guessing at the original intention, but I'd still prefer to move them all to top-level, I think it makes each of those features more visible that way and our top-level is not very crowded or anything currently.

**Comment by timpitman** [comment]
I agree - let's move these to the top level, with appropriate renames.
